integrate accelergy into scalesim for energy modeling

zishenwan opened the pull request:
Integrate Accelergy (https://accelergy.mit.edu) into Scalesim for energy modeling

Commit:
integrate accelergy into scalesim for energy modeling

Commit:
solve merge conflict

ritikraj7 committed:
Merge branch 'main' into scalesim-accelergy

ritikraj7 committed:
Update requirements.txt

added tk library for tinter

ritikraj7 committed:
Update testing-master.yml

ritikraj7 commented:
Hi @zishenwan, can you please update the readme on how to use accelergy integration. Thanks!

ritikraj7 committed:
Update requirements.txt

Added setuptools 
[setuptools is not a default package for python 3.12 onwards]

Commit:
updating golden trace due to a very small numerical mismatch in python 3.12 and prev versions

Commit:
pdate Accelergy integration and add README

ritikraj7 committed:
Merge branch 'main' into scalesim-accelergy

ritikraj7 committed:
Update testing-master.yml

Relaxing pylint score to 7.4 for development purpose, will raise it back to 7.5 before the release

ritikraj7 commented:
Closing this PR as it's superceded by https://github.com/scalesim-project/scale-sim-v2/pull/134

ritikraj7 commented:
Hi @agarwal-ayushi, please check the README https://github.com/scalesim-project/scale-sim-v3/blob/main/README_accelergy.md.

There was some problem with integration, but we recently fixed it in a different repo (https://github.com/scalesim-project/scale-sim-v3 repository). Please give it a try. You can also check the details of isssue and fixes here: https://github.com/scalesim-project/scale-sim-v3/pull/7.
